docs/vocabulary.md

```diff
@@ -388,6 +389,230 @@ pipeline in corresponding documentation.
+* Semantics: Create a request in another system; for example, create a Change Request that must be approved for a deployment to production.
+* Aliases: Approval, Ticket, Issue, Work Item
+* Inputs: Source Code Reference (Repository, Branch, Commit), Link to Previous Work Item, Provisioned Resources, Routes to Deployments
+* Outputs: Request Return Code/Results, Link to New Request
+* Other Results and Side Effects: Request created.
+
+#### Update Record
+* Semantics: Update a record in another system; for example, update and close a Change Request after a deployment to production; or, update a Github issue with the results of a policy check.
+* Aliases: Editor
+* Inputs: Source Code Reference (Repository, Branch, Commit), Link to Previous Request, Provisioned Resources, Routes to Deployments
+* Outputs: Update Return Code/Results, Link to Request
+* Other Results and Side Effects: Request record updated.
+
+#### Run
+* Semantics: Run a script or program that doesn't fall into one of the other categories. Often executes in another container.
+* Aliases: Execute
+* Inputs: Software Source, Binary Source, Generated Software, Generated Binaries, Packaged Artifacts, Provisioned Resources, Routes to Deployments
+* Outputs: Output or results of the script or program.
+* Other Results and Side Effects: Whatever the script or program has done.
+
+#### Record Results
+* Semantics: Record and report pipeline results and compliance evidence. Store pipeline artifacts for long-term archival.
+* Aliases: Audit, Attestation, Evidence, Report
+* Inputs: Any
+* Outputs: Compliance Reports, Archive Files, Return Code/Results
+* Other Results and Side Effects: Results, logs, reports, compliance evidence, and other pipeline artifacts are uploaded and/or archived.
+
+#### Cleanup
+* Semantics: Release build resources, de-provision environments, delete build workspace and build container(s).
+* Aliases: Finalize, Finish
+* Inputs: Build Environment, Build Workspace, Provisioned Resources, Routes to Deployments
+* Outputs: Return Code/Results
+* Other Results and Side Effects: Deleted Build Environment and Build Workspace, De-provisioned Resources, Deleted Deployments
+
+#### Any Step
+* Semantics: Some inputs and outputs are used across any and all stages.
+* Aliases: N/A
+* Inputs: Secrets, Build Environment, Build Workspace, Parameters for the Step
+* Outputs: Return Codes, Results, Records and Reports, Logs
+* Other Results and Side Effects: N/A
+
+#### Inputs and Outputs for Pipeline Steps
+
+The inputs and outputs listed in this table are the ones that are used by more than one step. These inputs and outputs need to be in a location and/or format that the pipeline steps expect so they can be processed correctly.
+
+With the exception of the Setup and Cleanup steps, all steps have the following inputs: Secrets, Build Environment, Build Workspace, Parameters for the Step; and the following outputs: Return Codes, Results, Records and Reports, Logs.
+
```

Comment:
Was there any idea to map these steps to the stages described in https://github.com/cdfoundation/sig-interoperability/pull/76?

Reply:
The general idea is that the steps map into stages, although it's definitely not 1-1 in practice.  We could have a separate doc somewhere with typical examples of different types of pipelines and pipeline stages.  I don't think it would live in the vocabulary.md document, though.  Maybe more of a CDF Best Practices SIG document?